src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -464,23 +465,45 @@ public void analyse(Network network, List<PropagatedContingency> contingencies,
 
                 // process contingencies with no connectivity break
                 for (PropagatedContingency contingency : connectivityBreakAnalysisResults.nonBreakingConnectivityContingencies()) {
+                    matrixShallowCopy(baseFlowStates, flowStates);
+                    matrixShallowCopy(baseFactorStates, factorStates);
+
                     calculateSensitivityValuesForAContingency(loadFlowContext, lfParametersExt, validFactorHolder, factorGroups,
-                            factorsStates, connectivityBreakAnalysisResults.contingenciesStates(), flowStates, contingency,
+                            factorStates, connectivityBreakAnalysisResults.contingenciesStates(), flowStates, contingency,
                             connectivityBreakAnalysisResults.contingencyElementByBranch(), Collections.emptySet(), participatingElements, Collections.emptySet(), resultWriter, reportNode, Collections.emptySet(), false);
                 }
 
                 LOGGER.info("Processing contingencies with connectivity break");
 
                 // process contingencies with connectivity break
                 for (ConnectivityBreakAnalysis.ConnectivityAnalysisResult connectivityAnalysisResult : connectivityBreakAnalysisResults.connectivityAnalysisResults()) {
+                    matrixShallowCopy(baseFlowStates, flowStates);
+                    matrixShallowCopy(baseFactorStates, factorStates);
+
                     processContingenciesBreakingConnectivity(connectivityAnalysisResult, loadFlowContext, lfParameters, lfParametersExt,
                             validFactorHolder, factorGroups, participatingElements, connectivityBreakAnalysisResults.contingencyElementByBranch(),
-                            flowStates, factorsStates, connectivityBreakAnalysisResults.contingenciesStates(), resultWriter, reportNode);
+                            flowStates, factorStates, connectivityBreakAnalysisResults.contingenciesStates(), resultWriter, reportNode);
                 }
             }
 
             stopwatch.stop();
             LOGGER.info("DC sensitivity analysis done in {} ms", stopwatch.elapsed(TimeUnit.MILLISECONDS));
         }
     }
+
+    /**
+     * Copy all the values that are in an originalMatrix and paste it in the copyMatrix (without allocating new memory spaces)
+     * The dimensions of both matrices must be the same
+     */
+    // TODO : implement this method for DenseMatrix in powsybl-core ?
+    private static void matrixShallowCopy(DenseMatrix originalMatrix, DenseMatrix copyMatrix) {
+        if (originalMatrix.getRowCount() == copyMatrix.getRowCount() && originalMatrix.getColumnCount() == copyMatrix.getColumnCount()) {
```

Comment:
The term shallow copy is incorrect. See definition here https://developer.mozilla.org/en-US/docs/Glossary/Shallow_copy
In a shallow copy in a shallow copy when you change either the source or the copy, you may also cause the other object to change too.

Reply:
Should we speak about "deep copy" ? (but without creating a new container)

Reply:
copyValues light be a bettre term and make clear you expect the same matrix dimensions.